main.go

```diff
@@ -491,6 +491,48 @@ func main() {
 			}
 		})
 
+		http.HandleFunc("/get-url", func(w http.ResponseWriter, r *http.Request) {
+			queryParameters := r.URL.Query()
+			var targetUrl string
+			if queryParameters.Has("url") {
+				targetUrl = queryParameters.Get("url")
+			} else {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+
+			resp, err := http.Get(targetUrl)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Request to target failed: %v", err), http.StatusBadGateway)
+				return
+			}
+			defer resp.Body.Close()
+
+			if r.Method == http.MethodHead {
+				// Just return the status code
+				w.WriteHeader(resp.StatusCode)
+				return
+			}
+
+			if r.Method == http.MethodGet {
+				// Copy headers
+				for key, values := range resp.Header {
+					for _, v := range values {
+						w.Header().Add(key, v)
+					}
+				}
+
+				// Set status code and copy body
+				w.WriteHeader(resp.StatusCode)
+				_, _ = io.Copy(w, resp.Body)
+				return
+			}
+
+			// Other methods not allowed
+			http.Error(w, "Only GET and HEAD are supported", http.StatusMethodNotAllowed)
```

Comment:
You should move this up to before you do the outbound GET call. Especially with above suggestion being applied. But in general, in an HTTP call handler, first validate everything on the request, then perform any logic.